Add error handling for HTTP requests in CoinsServiceProxy

sebsop opened the pull request:
Add error handling for HTTP requests in CoinsServiceProxy

Implemented try-catch blocks in methods for better error handling.
This enhancement allows the application to log exceptions and improves resilience during API interactions.

sebsop committed:
Add error handling for HTTP requests in CoinsServiceProxy

Implemented try-catch blocks in methods for better error handling.
This enhancement allows the application to log exceptions and improves resilience during API interactions.

sebsop committed:
Refactor CoinsService for error handling and documentation

- Updated constructor to use initializer for serviceProxy.
- Enhanced XML documentation for GetCoinBalanceAsync.
- Added try-catch blocks to handle exceptions in key methods.
- Included #pragma directives to suppress specific warnings.

sebsop committed:
Refactor CoinsServiceProxy constructor

Updated the constructor of the `CoinsServiceProxy` class to directly initialize the `httpClient` field with the provided parameter. Removed the old constructor and added remarks for better documentation of the class's purpose and parameters.

sebsop committed:
Enhance error handling in CoinsController

Updated the CoinsController to include try-catch blocks for error handling in methods related to user coin balance retrieval, spending coins, adding coins, and applying daily login bonuses. Exceptions are now logged, and appropriate HTTP status codes (500 for errors, 400 for already claimed bonuses) are returned. These changes improve the API's robustness and user experience.

sebsop committed:
Add error handling to repository async methods

Introduce try-catch blocks in several asynchronous methods
within the `Duo.Api.Repositories` namespace. Methods
`GetUserCoinBalanceAsync`, `TryDeductCoinsFromUserWalletAsync`,
`AddCoinsToUserWalletAsync`, `GetUserLastLoginTimeAsync`,
and `UpdateUserLastLoginTimeToNowAsync` now handle exceptions
during database operations, logging errors and returning
default values or re-throwing exceptions as needed. This
improves code robustness and error reporting.

UjfalusiAbel commented on Duo/Services/CoinsService.cs:
```diff
@@ -1,32 +1,35 @@
+            }
+            catch (Exception ex)
+            {
+                // Log the exception (e.g., to a logging service)
```
Delete unnecessary comments please

UjfalusiAbel commented on Duo/Services/CoinsService.cs:
```diff
@@ -1,32 +1,35 @@
+            {
+                // Log the exception (e.g., to a logging service)
+                Console.Error.WriteLine($"Error in GetCoinBalanceAsync: {ex.Message}");
+                return 0; // Return a default value
```
Delete unnecessary comments please

UjfalusiAbel commented on Duo/Services/CoinsService.cs:
```diff
@@ -47,7 +59,15 @@ public async Task<bool> TrySpendingCoinsAsync(int userId, int cost)
+            }
+            catch (Exception ex)
+            {
+                // Log the exception
```
Delete unnecessary comments please

UjfalusiAbel commented on Duo.Api/Repositories/Repository.cs:
```diff
@@ -1026,14 +1034,22 @@ public async Task<int> GetUserCoinBalanceAsync(int userId)
+                    await context.SaveChangesAsync();
+                    return true;
+                }
+                return false; // Not enough coins or user not found
```
Please delete unnecessary comments

UjfalusiAbel reviewed (changes requested):
Delete unnecessary comments

sebsop committed:
Clean up exception handling in CoinsService.cs

Removed commented-out logging statements from exception
handlers while preserving return statements. This enhances
code clarity without altering functionality.

sebsop committed:
Remove comments from exception handling in CoinsServiceProxy

This commit removes comments explaining return statements in the catch blocks of the `CoinsServiceProxy.cs` file. The functionality remains unchanged, but the code is simplified, which may impact clarity for future maintainers.

sebsop committed:
Refactor error handling in Repository.cs

Removed redundant comments from error handling logic in
multiple methods, streamlining the code while preserving
functionality. Return statements remain unchanged.
